akka-stream/src/main/scala/akka/stream/impl/StreamLayout.scala

```diff
@@ -300,6 +300,7 @@ private[stream] object VirtualProcessor {
   case object Completed extends Termination
   case class Failed(ex: Throwable) extends Termination
 
+  // FIXME this is not a legal Subscriber in the RS sense as a Subscriber may never be attached to more than 1 Publisher
```

Comment:
Good question!

I think I added it there as a reminder that this needs to be discussed.
I think the solution is to drop it and use a CancellingSubscriber instead.